odoo/addons/base/transpiler/js_transpiler.py

```diff
@@ -0,0 +1,594 @@
+
+        // Not a relative path:
+        // before
+        require("other_alias")
+        // after
+        require("other_alias")
+    """
+    new_content = content
+    for open_quote, path, close_quote in RELATIVE_REQUIRE_RE.findall(new_content):
+        if path.startswith(".") and "/" in path:
+            new_content = re.sub(rf"require\({str(open_quote)}{path}{str(close_quote)}\)", f'require("{relative_path_to_module_path(url, path)}")', new_content)
+    return new_content
+
+
+IMPORT_STAR = re.compile(r"^(?P<space>\s*)import\s+\*\s+as\s+(?P<identifier>\w+)\s*from\s*(?P<path>[^;\n]+)", re.MULTILINE | re.VERBOSE)
+
+
+def convert_star_import(content):
+    """
+    Transpile import star.
+
+    .. code-block:: javascript
+
+        // before
+        import * as name from "some/path"
+        // after
+        const name = require("some/path")
+    """
+    repl = r"\g<space>const \g<identifier> = require(\g<path>)"
+    return IMPORT_STAR.sub(repl, content)
+
+
+IMPORT_UNNAMED_RELATIVE_RE = re.compile(r"^(?P<space>\s*)import\s+(?P<path>[^;\n]+)", re.MULTILINE | re.VERBOSE)
+
+
+def convert_unnamed_relative_import(content):
+    """
+    Transpile relative "direct" imports. Direct meaning they are not store in a variable.
+
+    .. code-block:: javascript
+
+        // before
+        import "some/path"
+        // after
+        require("some/path")
+    """
+    repl = r"require(\g<path>)"
+    return IMPORT_UNNAMED_RELATIVE_RE.sub(repl, content)
+
+
+URL_INDEX_RE = re.compile(r"require\s*\(\s*(?P<path>(\".*/index/?\")|('.*/index/?')|(`.*/index/?`))\s*\)", re.MULTILINE | re.VERBOSE)
+
+
+def remove_index(content):
+    """
+    Remove in the paths the /index.js.
+    We want to be able to import a module just trough its directory name if it contains an index.js.
+    So we no longer need to specify the index.js in the paths.
+    """
+    def repl(matchobj):
+        path = matchobj["path"]
+        new_path = path[: path.rfind("/index")] + path[0]
+        return f"require({new_path})"
+    return URL_INDEX_RE.sub(repl, content)
+
+
+def relative_path_to_module_path(url, path_rel):
+    """
+    Convert the relative path into a module path, which is more generic and fancy.
+
+    :param path_rel: a relative path to the current url.
+    :return: module path (@module/...)
+    """
+    url_split = url.split("/")
+    path_rel_split = path_rel.split("/")
+    nb_back = len([v for v in path_rel_split if v == ".."]) + 1
+    result = "/".join(url_split[:-nb_back] + [v for v in path_rel_split if not v in ["..", "."]])
+    return url_to_module_path(result)
+
+
+ODOO_MODULE_RE = re.compile(r"""
+    \/(\*|\/).*\s+              # // or /*
+    @odoo-module\s+             # @odoo-module
+    (alias=(?P<alias>\S+))?\s+  # alias=web.AbstractAction (optional)
+    (default=(?P<default>\S+))? # default=False (optional)
+    """, re.VERBOSE)
+
+
+def is_odoo_module(content):
+    """
+    Detect if the file need to be transpiled.
```

Comment:
No, it does not.  The function determines if the content is likely an Odoo module, period.
Using its result to transpile or not is part of the asset's logic.

Reply:
Maybe "Detect if the file is a native odoo module"?